src/org/openstreetmap/josm/data/osm/NodeGraph.java

```diff
@@ -108,55 +136,124 @@ public static NodeGraph createUndirectedGraphFromNodeList(List<NodePair> pairs)
 
     /**
      * Create an undirected graph from the given ways, but prevent reversing of all
-     * non-new ways by fix one direction.
+     * non-new ways by fixing one direction.
      * @param ways Ways to build the graph from
      * @return node graph structure
      * @since 8181
      */
     public static NodeGraph createUndirectedGraphFromNodeWays(Collection<Way> ways) {
         NodeGraph graph = new NodeGraph();
-        for (Way w: ways) {
-            graph.add(buildNodePairs(w, false /* undirected */));
+        for (Way w : ways) {
+            graph.add(buildNodePairs(w, false));
         }
         return graph;
     }
 
+    /**
+     * Create a nearly undirected graph from the given ways, but prevent reversing of all
+     * non-new ways by fixing one direction.
+     * The first new way gives the direction of the graph.
+     * @param ways Ways to build the graph from
+     * @return node graph structure
+     */
     public static NodeGraph createNearlyUndirectedGraphFromNodeWays(Collection<Way> ways) {
         boolean dir = true;
         NodeGraph graph = new NodeGraph();
-        for (Way w: ways) {
+        for (Way w : ways) {
             if (!w.isNew()) {
                 /* let the first non-new way give the direction (see #5880) */
                 graph.add(buildNodePairs(w, dir));
                 dir = false;
             } else {
-                graph.add(buildNodePairs(w, false /* undirected */));
+                graph.add(buildNodePairs(w, false));
             }
         }
         return graph;
     }
 
-    private final Set<NodePair> edges;
-    private int numUndirectedEges;
-    /** counts the number of edges that were added */
-    private int addedEdges;
-    private final Map<Node, List<NodePair>> successors = new LinkedHashMap<>();
-    private final Map<Node, List<NodePair>> predecessors = new LinkedHashMap<>();
+    /**
+     * Add a node pair.
+     * @param pair node pair
+     */
```

Comment:
Wait a sec, this and most of the methods exist in master, I've just reordered the file to be more logical. Would you still like me to add the since xxx parts?

Reply:
No; I read through the new bits of the patch top-to-bottom. I didn't check to see what had moved around (I usually try avoid moving functions around -- it can make traversing history a PITA).

Reply:
Soo, should I revert the reordering?

Reply:
I would, especially if you didn't change anything besides the order. I wouldn't care if `git` was smart enough to follow reordering.

Reply:
Reordering also breaks the annotations feature:
https://josm.openstreetmap.de/browser/josm/trunk/src/org/openstreetmap/josm/data/osm/NodeGraph.java?annotate=blame

General rule: Do not change formatting (or order) of the code except for code directly affected by other changes. If formatting is broken in a major way and needs to be fixed, then do this independent from any other changes as a separate ticket.

Reply:
Alright, GerdP has a patch for this file in a different ticket anyway, and it would break that as well.

Reply:
Reordering doesn't _break_ the annotations/blame feature, but it sure makes it harder to use. There have been times when I've gone back in history and the blame is for a commit that changed whitespace. I _can_ work around that, but it takes time (I have to set the revision in the URL).

When figuring out why something is the way it is, `git blame`/`svn blame` are your friend, and formatting changes typically make life harder.

Reply:
Well. That's what I meant with break. You have a discontinuity (also called break :-) in the history and need to work around it.

Reply:
In 5e6e90b I changed back to the original order.